From: Divy Le Ray <divy@chelsio.com>

Protect against invalid phy entries in the eeprom.
Extend eeprom access timeout.

Signed-off-by: Divy Le Ray <divy@chelsio.com>
---

 drivers/net/cxgb3/t3_hw.c |    8 +++++++-
 1 files changed, 7 insertions(+), 1 deletions(-)

diff --git a/drivers/net/cxgb3/t3_hw.c b/drivers/net/cxgb3/t3_hw.c
index 968f64b..9a0898b 100644
--- a/drivers/net/cxgb3/t3_hw.c
+++ b/drivers/net/cxgb3/t3_hw.c
@@ -522,7 +522,7 @@ struct t3_vpd {
 	u32 pad;		/* for multiple-of-4 sizing and alignment */
 };
 
-#define EEPROM_MAX_POLL   4
+#define EEPROM_MAX_POLL   40
 #define EEPROM_STAT_ADDR  0x4000
 #define VPD_BASE          0xc00
 
@@ -3626,6 +3626,12 @@ int t3_prep_adapter(struct adapter *adap
 			++j;
 
 		p->port_type = &port_types[adapter->params.vpd.port_type[j]];
+ 		if (!p->port_type->phy_prep) {
+ 			CH_ALERT(adapter, "Invalid port type index %d\n",
+ 				 adapter->params.vpd.port_type[j]);
+ 			return -EINVAL;
+ 		}
+ 
 		p->port_type->phy_prep(&p->phy, adapter, ai->phy_base_addr + j,
 				       ai->mdio_ops);
 		mac_prep(&p->mac, adapter, j);
